pkgs/development/libraries/ffmpeg/generic.nix

```diff
@@ -334,8 +334,9 @@ stdenv.mkDerivation rec {
   pname = "ffmpeg" + (if ffmpegVariant == "minimal" then "" else "-${ffmpegVariant}");
   inherit version;
 
-  src = fetchurl {
-    url = "https://www.ffmpeg.org/releases/ffmpeg-${version}.tar.bz2";
+  src = fetchgit {
```

Comment:
Yes, it is more desirable to build from the actual source not the source + whatever manual steps? + compressed into a tarball.

I've had to try a different rev at least once already.

Asked the other way around, why should we keep using the tarballs?

Reply:
> Asked the other way around, why should we keep using the tarballs?

First, I am not suggesting that tarballs should be mandatory. I am merely asking why the trending changed here specifically.

Second, usually the tarballs need less dependencies: a fetchurl/curl versus fetchgit/git to begin with.

Third, sometimes the tarballs are preprocessed in order to simplify the building. Indeed in Nixpkgs we have `autoreconfHook` because sources grabbed from Git regularly expose its building scripts.

And fourth, this is irrelevant, but since you ask, here is your answer.